src/rand.rs

```diff
@@ -21,9 +22,100 @@ impl Distribution<Decimal> for Standard {
     }
 }
 
-#[test]
-fn has_random_decimal_instances() {
-    let mut rng = rand::rngs::OsRng;
-    let random: [Decimal; 32] = rng.gen();
-    assert!(random.windows(2).any(|slice| { slice[0] != slice[1] }));
+impl SampleUniform for Decimal {
+    type Sampler = DecimalSampler;
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct DecimalSampler {
+    mantissa_sampler: UniformInt<i128>,
+    scale: u32,
+}
+
+impl UniformSampler for DecimalSampler {
+    type X = Decimal;
+
+    #[inline]
+    fn new<B1, B2>(low: B1, high: B2) -> Self
+    where
+        B1: SampleBorrow<Self::X> + Sized,
+        B2: SampleBorrow<Self::X> + Sized,
+    {
+        let high = *high.borrow();
+        let high = Decimal::from_i128_with_scale(high.mantissa() - 1, high.scale());
+        UniformSampler::new_inclusive(low, high)
+    }
+
+    #[inline]
+    fn new_inclusive<B1, B2>(low: B1, high: B2) -> Self
+    where
+        B1: SampleBorrow<Self::X> + Sized,
+        B2: SampleBorrow<Self::X> + Sized,
+    {
+        let mut low = *low.borrow();
+        let mut high = *high.borrow();
+
+        // Set scales to match one another, because we are relying on mantissas'
+        // being comparable in order outsource the actual sampling implementation.
+        low.rescale(low.scale().max(high.scale()));
+        high.rescale(low.scale().max(high.scale()));
+
+        // Return our sampler, which contains an underlying i128 sampler so we
+        // outsource the actual randomness implementation.
+        Self {
+            mantissa_sampler: UniformInt::new_inclusive(low.mantissa(), high.mantissa()),
+            scale: low.scale(),
+        }
+    }
+
+    #[inline]
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Self::X {
+        let mantissa = self.mantissa_sampler.sample(rng);
+        Decimal::from_i128_with_scale(mantissa, self.scale)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashSet;
+
+    use super::*;
+
+    macro_rules! dec {
+        ($e:expr) => {
+            Decimal::from_str_exact(stringify!($e)).unwrap()
```

Comment:
I tried that and it didn't work. I think it only works for doctests and benchmarks and the like (where the `rust_decimal_macros` crate is external). I'll try again just in case, because I agree this is goofy.

Reply:
I just tried again.

    error: failed to resolve: could not find `rust_decimal` in the list of imported crates

 I think it works in doctests and the like but not unit tests. :-(

Reply:
Perhaps something like `use crate as rust_decimal;` could be possible if `rust_decimal_macros` didn't expand to absolute paths like `::rust_decimal`.

On a second thought, I am not sure if that would also be possible because `rust_decimal_macros` imports `rust_decimal`.

Reply:
Yup, and therein lies the problem.